src/browser/stacktrace/utils.ts

```diff
@@ -59,19 +61,90 @@ export function captureRawStackFrames(filterFunc?: (...args: unknown[]) => unkno
     const rawFramesPosition = targetObj.stack.indexOf("\n") + 1; // crop out error message
 
     return targetObj.stack.slice(rawFramesPosition);
-}
+};
+
+/**
+ * @description
+ * Rank values:
+ *
+ * 0: Can't extract code snippet; useless
+ *
+ * 1: WebdriverIO internals: Better than nothing
+ *
+ * 2: Project internals: Better than WebdriverIO internals, but worse, than user code part
+ *
+ * 3: User code: Best choice
+ */
+export const FRAME_REELVANCE: Record<string, { value: number; matcher: (fileName: string) => boolean }> = {
+    repl: { value: 0, matcher: fileName => /^REPL\d+$/.test(fileName) },
+    nodeInternals: { value: 0, matcher: fileName => /^node:[a-zA-Z\-_]/.test(fileName) },
+    wdioInternals: { value: 1, matcher: fileName => fileName.includes("/node_modules/webdriverio/") },
+    projectInternals: { value: 2, matcher: fileName => fileName.includes("/node_modules/") },
+    userCode: { value: 3, matcher: () => true },
+} as const;
+
+export const getFrameRelevance = (frame: StackFrame): number => {
+    if ([frame.fileName, frame.lineNumber, frame.columnNumber].some(_.isUndefined)) {
+        return 0;
+    }
+
+    const fileName: string = softFileURLToPath(frame.fileName!);
+
+    for (const factor in FRAME_REELVANCE) {
+        if (FRAME_REELVANCE[factor].matcher(fileName)) {
+            return FRAME_REELVANCE[factor].value;
+        }
+    }
+
+    return 0;
+};
 
-export function applyStackFrames(error: Error, frames: RawStackFrames): Error {
+const getStackTraceRelevance = (error: Error): number => {
+    const framesParsed = ErrorStackParser.parse(error);
+
+    return framesParsed.reduce((maxRelevance, frame) => {
+        return Math.max(maxRelevance, getFrameRelevance(frame));
+    }, 0);
+};
+
+const createErrorWithStack = (stack: RawStackFrames, errorMessage = ""): Error => {
+    const newError = new Error(errorMessage);
+
+    newError.stack = getErrorTitle(newError) + "\n" + stack;
+
+    return newError;
+};
+
+const applyStackTrace = (error: Error, stack: RawStackFrames): Error => {
     if (!error || !error.message) {
         return error;
     }
 
-    error.stack = getErrorTitle(error) + "\n" + frames;
+    error.stack = getErrorTitle(error) + "\n" + stack;
 
     return error;
-}
+};
+
+export const applyStackTraceIfBetter = (error: Error, stack: RawStackFrames): Error => {
```

Comment:
Calculates current stacktrace relevance and stacktrace relevance of the next nested level call.
Apples the new one if it suites better

Reply:
> Why not just leave the old name - `applyStackTrace`?

Lets pretend you don't know implementation details. You call 
```ts
const error = ...
const myNewStack = ...

applyStackTrace(error, myNewStack);
```
What do you think `error.stack` is now equal to? According to `applyStackTrace` function name, of course it is "myNewStack". But turns out, it is not, and its even worse. It is "myNewStack" sometimes, but also sometimes original "error.stack"

So that's why it is called "applyStackTraceIfBetter". That way developer would not be ensured "error.stack" to be "myNewStack". He would get from this function name that "it only apples stacktrace if it is better"